tests/api/test_api.py

```diff
@@ -125,6 +125,66 @@ def test_spot_detail_post(self, determine_stadsdeel):
         del data["point"]
         self.assertTrue(Spot.objects.filter(**data).exists())
 
+    @mock.patch("api.serializers.SpotSerializer.determine_stadsdeel")
+    def test_spot_detail_spottype_schoolstraat_post(self, determine_stadsdeel):
+        """post a new spot using the schoolstraat spottype. expect it to be created."""
+        determine_stadsdeel.return_value = Spot.Stadsdelen.Centrum
+
+        url = reverse("spot-list")
+        data = {
+            "locatie_id": "456",
+            "spot_type": Spot.SpotType.schoolstraat,
+            "description": "Test spot schoolstraat",
+            "point": '{"type": "Point","coordinates": [4.9239023,52.3875655]}',
+            "actiehouders": "Actiehouders test schoolstraat",
+            "status": "voorbereiding",
+        }
+        response = self.write_client.post(url, data=data)
+        self.assertStatusCode(url, response, expected_status=201)
+
+        del data["point"]
+        self.assertTrue(Spot.objects.filter(**data).exists())
+
+    @mock.patch("api.serializers.SpotSerializer.determine_stadsdeel")
+    def test_spot_detail_spottype_gebiedslocatie_ivm_post(self, determine_stadsdeel):
+        """post a new spot using the gebiedslocatie_ivm spottype. expect it to be created."""
+        determine_stadsdeel.return_value = Spot.Stadsdelen.Centrum
+
+        url = reverse("spot-list")
+        data = {
+            "locatie_id": "457",
+            "spot_type": Spot.SpotType.gebiedslocatie_ivm,
+            "description": "Test spot gebiedslocatie_ivm",
+            "point": '{"type": "Point","coordinates": [4.9239023,52.3875655]}',
+            "actiehouders": "Actiehouders test gebiedslocatie_ivm",
+            "status": "voorbereiding",
+        }
+        response = self.write_client.post(url, data=data)
+        self.assertStatusCode(url, response, expected_status=201)
+
+        del data["point"]
+        self.assertTrue(Spot.objects.filter(**data).exists())
+
+    @mock.patch("api.serializers.SpotSerializer.determine_stadsdeel")
+    def test_spot_detail_spottype_vso_post(self, determine_stadsdeel):
+        """post a new spot using the vso spottype. expect it to be created."""
+        determine_stadsdeel.return_value = Spot.Stadsdelen.Centrum
+
+        url = reverse("spot-list")
+        data = {
+            "locatie_id": "458",
+            "spot_type": Spot.SpotType.vso,
+            "description": "Test spot vso",
+            "point": '{"type": "Point","coordinates": [4.9239023,52.3875655]}',
+            "actiehouders": "Actiehouders test vso",
+            "status": "voorbereiding",
+        }
+        response = self.write_client.post(url, data=data)
+        self.assertStatusCode(url, response, expected_status=201)
+
+        del data["point"]
+        self.assertTrue(Spot.objects.filter(**data).exists())
+
```

Comment:
These testcases (and the one that already existed, `test_spot_detail_post`) are practically the same code. You should follow the DRY principle. In this case I do want to test all possible spot types, but the code itself should be a single function. 

To do so you can use parametrization. See https://docs.pytest.org/en/7.1.x/how-to/parametrize.html

Reply:
The paramertrize decorator of pytest doesn't work with unittest. I tried it and it didn't work and then I saw that according to https://pytest.org/en/7.1.x/how-to/unittest.html this is not supported. There are many solutions, like using an external library parameterized, or creating a separate class that doesn't inherit from TestCase. But the cleanest one I saw was just create a list of tuple of objets and loop through it, the same what the decorator does at the end. If you suggest a better solution, let me please know.